pkg/engine/target/character/execute.go

```
@@ -83,6 +83,10 @@ func (mgr *Manager) ExecuteAction(id key.TargetID, isInsert bool) (target.Execut
 		return target.ExecutableAction{}, err
 	}
 
+	canUseAttack, err := mgr.engine.CanUseAttack(id)
+	if err != nil || !canUseAttack {
+		return target.ExecutableAction{}, err
+	}
```

might want to return a custom error message somewhere in case both `canUseSkill` and `canUseAttack` fail (which should never happen ingame)